code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -764,25 +790,48 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 	else
 		return 0
 
-/obj/machinery/chem_master/condimaster
-	name = "CondiMaster 3000"
-	condi = 1
-
 /obj/machinery/chem_master/update_icon()
+
+	overlays.len = 0
+
 	if(beaker)
 		icon_state = "mixer1"
+
+		if(beaker.reagents.total_volume)
+			var/image/filling = image('icons/obj/reagentfillings.dmi', src, "mixbeaker")
+
+			var/percent = round((beaker.reagents.total_volume / beaker.volume) * 100)
+			switch(percent)
+				if(0 to 24) 	filling.icon_state += "10" //Intentionally compressed
+				if(25 to 49)	filling.icon_state += "25"
+				if(50 to 74)	filling.icon_state += "50"
+				if(75 to 79)	filling.icon_state += "75"
+				if(80 to 90)	filling.icon_state += "80"
+				if(91 to INFINITY)	filling.icon_state += "100"
+
+			filling.icon += mix_color_from_reagents(beaker.reagents.reagent_list)
+			overlays += filling

```

Comment:
The problem is that it would not work because there's two prongs holding the beaker in place, and the beaker itself is downscaled by about one pixel in every direction